### PR TITLE
update the user table & associations & validation

### PR DESCRIPTION
**Pull request by lady-rosie**
create a full name method and add validations and associations to user model

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -3,4 +3,23 @@ class User < ApplicationRecord
...
+
+  #users must specify their first and last name
+  validates :first_name, presence: true
+  validates :last_name, presence: true
+
+  #user associations
+  has_many :gardens
+  has_many :tasks
+  has_many :messages
+  has_many :conversations, through: :user_conversations
```
Comment (jscouller09) [inline app/models/user.rb]: need to add the following before line 18 so the through association will work => has_many :user_conversations

File: app/models/user.rb [hunk 2]
```diff
@@ -3,4 +3,23 @@ class User < ApplicationRecord
...
+
+
+  #method to access user full name
+  def full_name
+    "#{first_name} #{last_name}"
```
Comment (jscouller09) [inline app/models/user.rb]: Can we add capitalize method to both names (if they entered lowercase?)

Comment (lady-rosie) [reply app/models/user.rb]: will do ;)

File: app/models/user.rb (hunk 1 above)
Comment (lady-rosie) [reply app/models/user.rb]: done

**Pull request merged**